src/scheduler/prime.cpp

```diff
@@ -241,19 +241,20 @@ is_holiday(long date)
 static void
 handle_missing_prime_info(void)
 {
-	enum days d;
+	//enum days d;
+	int d;
 
 	for (d = SUNDAY; d < HIGH_DAY; d++) {
 		if (conf.prime[d][PRIME].all + conf.prime[d][PRIME].none
 				+ conf.prime[d][PRIME].hour + conf.prime[d][PRIME].min == 0) {
 			conf.prime[d][PRIME].all = TRUE;
 			conf.prime[d][PRIME].none = FALSE;
-			conf.prime[d][PRIME].hour = UNSPECIFIED;
-			conf.prime[d][PRIME].min = UNSPECIFIED;
+			conf.prime[d][PRIME].hour = static_cast<unsigned int>(UNSPECIFIED);
+			conf.prime[d][PRIME].min = static_cast<unsigned int>(UNSPECIFIED);
 			conf.prime[d][NON_PRIME].none = TRUE;
 			conf.prime[d][NON_PRIME].all = FALSE;
-			conf.prime[d][NON_PRIME].hour = UNSPECIFIED;
-			conf.prime[d][NON_PRIME].min = UNSPECIFIED;
+			conf.prime[d][NON_PRIME].hour = static_cast<unsigned int>(UNSPECIFIED);
+			conf.prime[d][NON_PRIME].min = static_cast<unsigned int>(UNSPECIFIED);
```

Comment:
when we do this cast, it becomes UINT_MAX
should i replace these with UINT_MAX?

Reply:
No, UNSPECIFIED is not supposed to be used with '<' '>'.  It should be checked if a value is equal to it.  Because of that, it doesn't really matter if we're comparing -1 or UINT_MAX, as long as we're comparing it to UNSPECIFIED.  I just don't see any purpose to making these unsigned when they can be signed.  Like I said before, clang complains about this.  We might as well get rid of that warning.

Reply:
We do compare `hour` and `min` in end_prime_status_rec

https://github.com/vstumpf/pbspro/blob/2bccef5a9d23db8ebe8b6b7ff03128d2e9a58cfa/src/scheduler/prime.cpp#L603-L606

Reply:
Well that is interesting.  You found yourself a bug :).  In light of this, I agree with you.  We should leave this as unsigned.